addons/mail/static/src/models/user/user.js

```diff
@@ -195,25 +184,46 @@ function factory(dependencies) {
         }
 
         /**
-         * @override
+         * @private
+         * @returns {string}
          */
-        _updateAfter(previous) {
-            if (this.partnerDisplayName && this.partner) {
-                this.partner.update({ display_name: this.partnerDisplayName });
-            }
+        _computeDisplayName() {
+            return this.display_name || this.partner && this.partner.display_name;
         }
 
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeNameOrDisplayName() {
+            return this.display_name;
```

Comment:
`name` is the same on both models. Should use the one from partner in priority, just like the method did when it was not a compute.

```suggestion
            return this.partner && this.partner.nameOrDisplayName || this.display_name;
```